rclcpp/include/rclcpp/strategies/allocator_memory_strategy.hpp

```diff
@@ -78,7 +78,7 @@ class AllocatorMemoryStrategy : public memory_strategy::MemoryStrategy
   {
     for (auto it = guard_conditions_.begin(); it != guard_conditions_.end(); ++it) {
       if (*it == guard_condition) {
-        guard_conditions_.erase(it);
+        it = guard_conditions_.erase(it);
```

Comment:
Why is this necessary? The next line breaks the loop and leaves the scope of `it`. I would consider the newly assigned `it` unused.

Reply:
That's a good point, I over matched it, I think I fixed it up correctly here: e0be1f6